packages/medusa/src/services/event-bus.ts

```diff
@@ -268,15 +275,15 @@ export default class EventBusService {
    */
   worker_ = async <T>(job: BullJob<T>): Promise<unknown> => {
     const { eventName, data } = job.data
-    const eventObservers = this.observers_.get(eventName) || []
-    const wildcardObservers = this.observers_.get("*") || []
+    const eventObservers = this.eventToSubscribersMap_.get(eventName) || []
+    const wildcardObservers = this.eventToSubscribersMap_.get("*") || []
 
     const observers = eventObservers.concat(wildcardObservers)
 
     // Pull already completed subscribers from the job data
-    const completedSubscribers = job.data.completed_subscribers || []
+    const completedSubscribers = job.data.completed_subscriber_ids || []
 
-    // Filter out already completed subscribers from the all observers
+    // Filter out already completed subscribers from the all subscribers
     const subscribers = observers.filter(
```

Comment:
comment: descriptors here as well

Reply:
I don't think `descriptors` is necessary here.